pkg/controllers/work/apply_controller.go

```diff
@@ -426,6 +465,11 @@ func (r *ApplyWorkReconciler) patchCurrentResource(ctx context.Context, gvr sche
 		klog.ErrorS(patchErr, "failed to patch the manifest", "gvr", gvr, "manifest", manifestRef)
 		return nil, ManifestNoChangeAction, patchErr
 	}
+
+	// calculate manifest hash & last modified config again to set the new manifest hash & last applied config after patch gets applied
+	if err := r.updateConfigMap(ctx, curObjConfigMap, manifestObj); err != nil {
```

Comment:
Hi Arvind! Just to double-check, is it really necessary to update the configmap for a second time here?

Reply:
This need to be updated with the object from the hub cluster